tests/test_fgnhg.py

```diff
@@ -325,6 +326,45 @@ def test_route_fgnhg(self, dvs, testlog):
         program_route_and_validate_fine_grained_ecmp(app_db.db_connection, asic_db, state_db, ip_to_if_map,
                             fg_nhg_prefix, nh_memb_exp_count, nh_oid_map, nhgid, bucket_size)
 
+        #############################################################################
+        #                                                                           #
+        #                        swss Warm-Restart Testing Begin                    #
+        #                                                                           #
+        #############################################################################
+        dvs.runcmd("config warm_restart enable swss")
+
+        # Stop swss before modifing the configDB
+        dvs.stop_swss()
+        time.sleep(1)
+
+        # start to apply new port_config.ini
+        dvs.start_swss()
+        dvs.runcmd(['sh', '-c', 'supervisorctl start neighsyncd'])
+        dvs.runcmd(['sh', '-c', 'supervisorctl start restore_neighbors'])
+
+        # Enabling some extra logging for validating the order of orchagent
+        dvs.runcmd("swssloglevel -l INFO -c orchagent")
+        time.sleep(5)
+
+        keys = asic_db.wait_for_n_keys(ASIC_ROUTE_TB, asic_routes_count + 1)
+        nhgid = asic_route_exists_and_is_nhg(asic_db, keys, fg_nhg_prefix)
+        assert nhgid is not None
+
+        validate_asic_nhg(asic_db, nhgid, bucket_size)
+
+        nh_oid_map = get_nh_oid_map(asic_db)
+
+        nh_memb_exp_count = {"10.0.0.1":10,"10.0.0.3":10,"10.0.0.5":10,"10.0.0.7":10,"10.0.0.9":10,"10.0.0.11":10}
+        program_route_and_validate_fine_grained_ecmp(app_db.db_connection, asic_db, state_db, ip_to_if_map,
+                            fg_nhg_prefix, nh_memb_exp_count, nh_oid_map, nhgid, bucket_size)
+
+        #############################################################################
+        #                                                                           #
+        #                        swss Warm-Restart Testing END                      #
+        #                            get to normal state                            #
+        #############################################################################
+
```

Comment:
Could we add a warm reboot test case which utilizes the inactive_to_active_map, to make sure that a case where all members in a bank are down will work fine after warm reboot in terms of in-memory state? 

We can do the following for the test case:
1. Create Fine Grained ECMP route with only members of 1 bank present(...7, ...9, ...11)
2. Run warm reboot
3. Check everything is same post warm reboot
4. Add a route which changes members in 1 bank, eg: (...9, ...11), check everything is configured fine
5. Add a route which adds a member in a previously down bank, eg(...1, ...9, ...11), check everything is configured fine

Reply:
test warm reboot with only one bank up and vs test works.